src/CommandingOfficers/CommanderAbility.java

```diff
@@ -38,45 +39,72 @@ public String toString()
     return myName;
   }
 
-  /** Provide a hook to increase the ability's cost for its next invocation.
+  /** Provides a hook to increase the ability's cost for its next invocation.
    * Being in its own function allows an easy way for individual abilities
    * to change the cost function if needed.
    */
-  protected void adjustCost()
+  public void adjustCost()
   {
     // Increase the cost of this ability for next time to mitigate spam and
     // accommodate the presumably-growing battlefront.
     // Cost grows by at least one, and at most 10% of the current cost.
     myPowerCost += Math.max(myPowerCost*0.1, 1);
   }
 
-  /** Final method to do some bookkeeping, and then call
-   * perform() do do the actual work. This allows us to
-   * manage global Ability side-effects in one place. */
-  public final void activate(MapMaster gameMap)
+  /**
+   * Public hook to apply this Ability's effects to a given Commander.
+   */
+  public final void activate(Commander co, MapMaster gameMap)
+  {
+    // Don't re-apply CO mods if we've already applied them to this CO
+    if(!coModsApplied.containsKey(co)) {
+      ArrayList<COModifier> coModsToApply = new ArrayList<COModifier>();
+      enqueueCOMods(co, gameMap, coModsToApply);
+      coModsApplied.put(co, coModsToApply);
+      for(COModifier com : coModsToApply)
+        com.applyChanges(co);
+    }
+    perform(co, gameMap);
+  }
+  /**
+   * Public hook to handle cleanup.
+   * <p>One call per use of the power should be sufficient.
+   */
+  public final void deactivate(MapMaster gameMap)
   {
-    if( myCommander.getAbilityPower() < myPowerCost )
+    for( Commander co : new ArrayList<Commander>(coModsApplied.keySet()) )
     {
-      System.out.println("WARNING!: Performing ability with insufficient ability power!");
+      if( coModsApplied.containsKey(co) )
```

Comment:
Unnecessary conditional to check if `co` is part of the map it is part of (because we are iterating over it).

Reply:
I believe I've cleaned this up already.